From linux-usb-owner@vger.kernel.org  Wed Jul 21 12:18:08 2010
From: tom.leiming@gmail.com
To: greg@kroah.com, stern@rowland.harvard.edu
Cc: linux-usb@vger.kernel.org, Ming Lei <tom.leiming@gmail.com>
Subject: USB: serial: enable async suspend for usb serial port device
Date: Tue, 13 Jul 2010 23:56:24 +0800
Message-Id: <1279036584-3709-1-git-send-email-tom.leiming@gmail.com>

From: Ming Lei <tom.leiming@gmail.com>

Usb serial port device is child of its usb interface device, so
we can enable async suspend of usb serial port device to speedup
system suspend.

Signed-off-by: Ming Lei <tom.leiming@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/usb-serial.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/serial/usb-serial.c
+++ b/drivers/usb/serial/usb-serial.c
@@ -1071,6 +1071,8 @@ int usb_serial_probe(struct usb_interfac
 		dev_set_name(&port->dev, "ttyUSB%d", port->number);
 		dbg ("%s - registering %s", __func__, dev_name(&port->dev));
 		port->dev_state = PORT_REGISTERING;
+		device_enable_async_suspend(&port->dev);
+
 		retval = device_add(&port->dev);
 		if (retval) {
 			dev_err(&port->dev, "Error registering port device, "
